actors/miner/tests/vesting_unvested_funds.rs

```diff
@@ -0,0 +1,132 @@
+use fil_actor_miner::VestSpec;
+use fil_actor_miner::VestingFunds;
+use fvm_ipld_encoding::CborStore;
+use fvm_shared::bigint::Zero;
+use fvm_shared::econ::TokenAmount;
+
+mod state_harness;
+use state_harness::*;
+
+#[test]
+fn unlock_unvested_funds_leaving_bucket_with_non_zero_tokens() {
+    let mut h = StateHarness::new(0);
+    let vspec = VestSpec { initial_delay: 0, vest_period: 5, step_duration: 1, quantization: 1 };
+
+    let vest_start = 100;
+    let vest_sum = TokenAmount::from(100);
+
+    h.add_locked_funds(vest_start, &vest_sum, &vspec).unwrap();
+
+    let amount_unlocked = h.unlock_unvested_funds(vest_start, &TokenAmount::from(39)).unwrap();
+    assert_eq!(TokenAmount::from(39), amount_unlocked);
+
+    assert_eq!(TokenAmount::from(0), h.unlock_vested_funds(vest_start).unwrap());
+    assert_eq!(TokenAmount::from(0), h.unlock_vested_funds(vest_start + 1).unwrap());
```

Comment:
nit: it would help clarify what's going on a bit to add a comment above this unlock saying "// no vested funds available to unlock until strictly after first vesting epoch"